projects/rocprofiler-sdk/source/lib/python/rocpd/source/perfetto.cpp

```diff
@@ -517,6 +517,102 @@ write_perfetto(
         for(auto ditr : kernel_dispatch_gen)
         {
             auto gen = kernel_dispatch_gen.get(ditr);
+
+            // Temporary fix until timestamp issues are resolved: Set timestamps to be
+            // halfway between ending timestamp and starting timestamp of overlapping
+            // kernel dispatches. Perfetto displays slices incorrectly if overlapping
+            // slices on the same track are not completely enveloped.
+            {
+                struct kernel_dispatch_group_index
+                {
+                    uint64_t agent_abs_index;
```

Comment:
```suggestion
                    uint64_t agent_absolute_index;
```
Jonathan's 4.0 schema gets rid of all these "agent_abs_index" and call out the full name "agent_absolute_index" for consistency.  Since you're creating this struct, might as well adopt it now.

Reply:
Done